packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -116,20 +91,59 @@ export const getPriceData = (product, opts = {}) => {
     }
 }
 
+/**
+ * Find the lowest price of a product, across all of its price books and promotions
+ * @param {Object} product - product data from the API
+ * @returns {Object|undefined} the lowest price for the given product. If it's a promotional price, the promotion will also be returned.
+ */
+export const findLowestPrice = (product) => {
+    if (!product) return
+    const array = product.variants ?? [product]
+
+    return array.reduce(
+        (prev, data) => {
+            const promotions = data.productPromotions || []
+            const [smallestPromotionalPrice, promo] = getSmallestValByProperty(
+                promotions,
+                'promotionalPrice'
+            )
+
+            let salePrice, promotion
+            if (smallestPromotionalPrice && smallestPromotionalPrice < data.price) {
+                salePrice = smallestPromotionalPrice
+                promotion = promo
+            } else {
+                salePrice = data.price
+                promotion = null
+            }
```

Comment:
nit: to simplify the code
```suggestion
            let salePrice = data.price
            let promotion = null
            if (smallestPromotionalPrice && smallestPromotionalPrice < data.price) {
                salePrice = smallestPromotionalPrice
                promotion = promo
            } 
```